### ISSUE TITLE
IPA server install: Directory server creation prints Traceback in SetupDs.create_from_args

### ISSUE THREAD

**Issue by flo-renaud** [389ds/389-ds-base#4468]
**Issue Description**
During the installation of ipa server, a Traceback is diplayed when the call to SetupDs.create_from_args is called:
```
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib64/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 664, in format
    record.message = record.getMessage()
  File "/usr/lib64/python3.8/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
  File "/usr/lib/python3.8/site-packages/lib389/_entry.py", line 285, in __repr__
    ldif.LDIFWriter(
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 195, in unparse
    self._unparseAttrTypeandValue('dn', dn)
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 145, in _unparseAttrTypeandValue
    self._unfold_lines(': '.join([attr_type, attr_value.decode('ascii')]))
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 108, in _unfold_lines
    self._output_file.write(line)
TypeError: descriptor 'write' for '_io.StringIO' objects doesn't apply to a 'str' object
```

**Package Version and Platform:**
 - Platform: Fedora 32
 - Package and version: 389-ds-base-2.0.1-20201129gitce7beae4d.fc32.x86_64

**Steps to Reproduce**
1. call ipa installer with `ipa-server-install --domain $DOMAIN --realm $REALM -a $PASSWORD -p $PASSWORD -U`
2. the install succeeds but prints multiple Tracebacks

**Expected results**
We shouldn't see any Traceback in the output

**Additional context**
Logs are available in freeip nightly PR #[557](https://github.com/freeipa-pr-ci2/freeipa/pull/557) in the test directory:
- [report](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/9323274c-31bc-11eb-8437-fa163e6c3b05/report.html)
- [logs](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/9323274c-31bc-11eb-8437-fa163e6c3b05/test_integration-test_replication_layouts.py-TestStarTopologyWithoutCA-test_star_topology_without_ca/master.ipa.test/)

**Comment by flo-renaud** [comment]
The full traceback:
```
  [1/45]: creating directory server instance
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib64/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 664, in format
    record.message = record.getMessage()
  File "/usr/lib64/python3.8/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
  File "/usr/lib/python3.8/site-packages/lib389/_entry.py", line 285, in __repr__
    ldif.LDIFWriter(
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 195, in unparse
    self._unparseAttrTypeandValue('dn', dn)
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 145, in _unparseAttrTypeandValue
    self._unfold_lines(': '.join([attr_type, attr_value.decode('ascii')]))
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 108, in _unfold_lines
    self._output_file.write(line)
TypeError: descriptor 'write' for '_io.StringIO' objects doesn't apply to a 'str' object
Call stack:
  File "/usr/sbin/ipa-server-install", line 25, in <module>
    ipa_server_install.run()
  File "/usr/lib/python3.8/site-packages/ipaserver/install/ipa_server_install.py", line 51, in run
    ServerInstall.run_cli()
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 132, in run_cli
    sys.exit(cls.main(sys.argv))
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 156, in main
    return command.execute()
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 180, in execute
    return_value = self.run()
  File "/usr/lib/python3.8/site-packages/ipapython/install/cli.py", line 340, in run
    return cfgr.run()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 360, in run
    return self.execute()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 386, in execute
    for rval in self._executor():
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 421, in __runner
    step()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 418, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python3.8/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 655, in _configure
    next(executor)
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 421, in __runner
    step()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 418, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python3.8/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python3.8/site-packages/ipapython/install/common.py", line 65, in _install
    for unused in self._installer(self.parent):
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/__init__.py", line 575, in main
    master_install(self)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/install.py", line 272, in decorated
    func(installer)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/install.py", line 844, in install
    ds.create_instance(realm_name, host_name, domain_name,
  File "/usr/lib/python3.8/site-packages/ipaserver/install/dsinstance.py", line 318, in create_instance
    self.start_creation(runtime=30)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/service.py", line 603, in start_creation
    run_step(full_msg, method)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/service.py", line 589, in run_step
    method()
  File "/usr/lib/python3.8/site-packages/ipaserver/install/dsinstance.py", line 560, in __create_instance
    sds.create_from_args(general, slapd, backends, None)
  File "/usr/lib/python3.8/site-packages/lib389/instance/setup.py", line 674, in create_from_args
    self._install_ds(general, slapd, backends)
  File "/usr/lib/python3.8/site-packages/lib389/instance/setup.py", line 910, in _install_ds
    ds_instance.start(timeout=60)
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 1125, in start
    self.open()
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 1037, in open
    self.__add_brookers__()
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 356, in __add_brookers__
    self.monitorldbm = MonitorLDBM(self)
  File "/usr/lib/python3.8/site-packages/lib389/monitor.py", line 164, in __init__
    if not ds_is_older("1.4.0", instance=instance):
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1101, in ds_is_older
    return ds_is_related('older', *ver, instance=instance)
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1088, in ds_is_related
    ds_ver = get_ds_version(instance.ds_paths)
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1076, in get_ds_version
    return paths.version
  File "/usr/lib/python3.8/site-packages/lib389/paths.py", line 162, in __getattr__
    ent = self._instance.getEntry(dn, attrlist=[attr,])
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 1704, in getEntry
    self.log.debug("Retrieved entry %s", obj)
Unable to print the message and arguments - possible formatting error.
Use the traceback above to help find the error.
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib64/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 664, in format
    record.message = record.getMessage()
  File "/usr/lib64/python3.8/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
  File "/usr/lib/python3.8/site-packages/lib389/_entry.py", line 285, in __repr__
    ldif.LDIFWriter(
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 195, in unparse
    self._unparseAttrTypeandValue('dn', dn)
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 145, in _unparseAttrTypeandValue
    self._unfold_lines(': '.join([attr_type, attr_value.decode('ascii')]))
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 108, in _unfold_lines
    self._output_file.write(line)
TypeError: descriptor 'write' for '_io.StringIO' objects doesn't apply to a 'str' object
Call stack:
  File "/usr/sbin/ipa-server-install", line 25, in <module>
    ipa_server_install.run()
  File "/usr/lib/python3.8/site-packages/ipaserver/install/ipa_server_install.py", line 51, in run
    ServerInstall.run_cli()
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 132, in run_cli
    sys.exit(cls.main(sys.argv))
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 156, in main
    return command.execute()
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 180, in execute
    return_value = self.run()
  File "/usr/lib/python3.8/site-packages/ipapython/install/cli.py", line 340, in run
    return cfgr.run()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 360, in run
    return self.execute()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 386, in execute
    for rval in self._executor():
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 421, in __runner
    step()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 418, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python3.8/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 655, in _configure
    next(executor)
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 421, in __runner
    step()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 418, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python3.8/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python3.8/site-packages/ipapython/install/common.py", line 65, in _install
    for unused in self._installer(self.parent):
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/__init__.py", line 575, in main
    master_install(self)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/install.py", line 272, in decorated
    func(installer)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/install.py", line 844, in install
    ds.create_instance(realm_name, host_name, domain_name,
  File "/usr/lib/python3.8/site-packages/ipaserver/install/dsinstance.py", line 318, in create_instance
    self.start_creation(runtime=30)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/service.py", line 603, in start_creation
    run_step(full_msg, method)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/service.py", line 589, in run_step
    method()
  File "/usr/lib/python3.8/site-packages/ipaserver/install/dsinstance.py", line 560, in __create_instance
    sds.create_from_args(general, slapd, backends, None)
  File "/usr/lib/python3.8/site-packages/lib389/instance/setup.py", line 674, in create_from_args
    self._install_ds(general, slapd, backends)
  File "/usr/lib/python3.8/site-packages/lib389/instance/setup.py", line 911, in _install_ds
    ds_instance.open()
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 1037, in open
    self.__add_brookers__()
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 356, in __add_brookers__
    self.monitorldbm = MonitorLDBM(self)
  File "/usr/lib/python3.8/site-packages/lib389/monitor.py", line 164, in __init__
    if not ds_is_older("1.4.0", instance=instance):
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1101, in ds_is_older
    return ds_is_related('older', *ver, instance=instance)
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1088, in ds_is_related
    ds_ver = get_ds_version(instance.ds_paths)
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1076, in get_ds_version
    return paths.version
  File "/usr/lib/python3.8/site-packages/lib389/paths.py", line 162, in __getattr__
    ent = self._instance.getEntry(dn, attrlist=[attr,])
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 1704, in getEntry
    self.log.debug("Retrieved entry %s", obj)
Unable to print the message and arguments - possible formatting error.
Use the traceback above to help find the error.
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib64/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/usr/lib64/python3.8/logging/__init__.py", line 664, in format
    record.message = record.getMessage()
  File "/usr/lib64/python3.8/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
  File "/usr/lib/python3.8/site-packages/lib389/_entry.py", line 285, in __repr__
    ldif.LDIFWriter(
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 195, in unparse
    self._unparseAttrTypeandValue('dn', dn)
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 145, in _unparseAttrTypeandValue
    self._unfold_lines(': '.join([attr_type, attr_value.decode('ascii')]))
  File "/usr/lib64/python3.8/site-packages/ldif.py", line 108, in _unfold_lines
    self._output_file.write(line)
TypeError: descriptor 'write' for '_io.StringIO' objects doesn't apply to a 'str' object
Call stack:
  File "/usr/sbin/ipa-server-install", line 25, in <module>
    ipa_server_install.run()
  File "/usr/lib/python3.8/site-packages/ipaserver/install/ipa_server_install.py", line 51, in run
    ServerInstall.run_cli()
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 132, in run_cli
    sys.exit(cls.main(sys.argv))
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 156, in main
    return command.execute()
  File "/usr/lib/python3.8/site-packages/ipapython/admintool.py", line 180, in execute
    return_value = self.run()
  File "/usr/lib/python3.8/site-packages/ipapython/install/cli.py", line 340, in run
    return cfgr.run()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 360, in run
    return self.execute()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 386, in execute
    for rval in self._executor():
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 421, in __runner
    step()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 418, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python3.8/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 655, in _configure
    next(executor)
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 421, in __runner
    step()
  File "/usr/lib/python3.8/site-packages/ipapython/install/core.py", line 418, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python3.8/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python3.8/site-packages/ipapython/install/common.py", line 65, in _install
    for unused in self._installer(self.parent):
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/__init__.py", line 575, in main
    master_install(self)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/install.py", line 272, in decorated
    func(installer)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/server/install.py", line 844, in install
    ds.create_instance(realm_name, host_name, domain_name,
  File "/usr/lib/python3.8/site-packages/ipaserver/install/dsinstance.py", line 318, in create_instance
    self.start_creation(runtime=30)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/service.py", line 603, in start_creation
    run_step(full_msg, method)
  File "/usr/lib/python3.8/site-packages/ipaserver/install/service.py", line 589, in run_step
    method()
  File "/usr/lib/python3.8/site-packages/ipaserver/install/dsinstance.py", line 582, in __create_instance
    inst.open()
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 1037, in open
    self.__add_brookers__()
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 356, in __add_brookers__
    self.monitorldbm = MonitorLDBM(self)
  File "/usr/lib/python3.8/site-packages/lib389/monitor.py", line 164, in __init__
    if not ds_is_older("1.4.0", instance=instance):
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1101, in ds_is_older
    return ds_is_related('older', *ver, instance=instance)
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1088, in ds_is_related
    ds_ver = get_ds_version(instance.ds_paths)
  File "/usr/lib/python3.8/site-packages/lib389/utils.py", line 1076, in get_ds_version
    return paths.version
  File "/usr/lib/python3.8/site-packages/lib389/paths.py", line 162, in __getattr__
    ent = self._instance.getEntry(dn, attrlist=[attr,])
  File "/usr/lib/python3.8/site-packages/lib389/__init__.py", line 1704, in getEntry
    self.log.debug("Retrieved entry %s", obj)
Unable to print the message and arguments - possible formatting error.
Use the traceback above to help find the error.
  [2/45]: configure autobind for root
  [3/45]: tune ldbm plugin
```

**Comment by mreynolds389** [comment]
This is a regression from:    https://github.com/389ds/389-ds-base/pull/4456

**Comment by mreynolds389** [comment]
@flo-renaud this should be fixed now in 389-ds-base-2.0.1-20201201gitb7219518d.fc32

**Comment by flo-renaud** [comment]
@mreynolds389 
verified in our tests PR#[574](https://github.com/freeipa-pr-ci2/freeipa/pull/574), with 389-ds-base-2.0.1-20201205gitdec149b4c.fc32.x86_64. The error message is not present anymore, issue can be closed.
